src/core/server/http/router/request.ts

```diff
@@ -22,10 +22,26 @@ import { ObjectType, TypeOf } from '@kbn/config-schema';
 import { Request } from 'hapi';
 
 import { filterHeaders, Headers } from './headers';
-import { RouteSchemas } from './route';
+import { RouteSchemas, RouteConfigOptions } from './route';
+
+interface KibanaRequestRoute {
+  path: string;
+  method: 'get' | 'post' | 'put' | 'patch' | 'delete' | 'options';
+  options: RouteConfigOptions;
+}
 
 /** @public */
 export class KibanaRequest<Params = unknown, Query = unknown, Body = unknown> {
+  public static getRoute(request: Request): KibanaRequestRoute {
```

Comment:
__note:__ Why do we need this method to be public? Feels like `KibanaRequest` is a part of public API that will be extensively used by plugins, so better we expose only what we commit to maintain backward compatibility for.

Also `getRoute` could be an instance method on `KibanaRequest` since it gets data only from `req` that's already instance property. But it depends on how we see this method being used of course, so up to you.